### PR TITLE
fix: Refetch galleries For You if no results for followed artists

### PR DESCRIPTION
**Pull request by olerichter00**
### Description

We need to refetch the "Galleries For You" query if there are no results. This means that the user probably doesn't follow any artists that have artworks listed by a gallery.

In that case we refetch the query (once) without filtering for galleries listing artworks by artists the user follows
### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/GalleriesForYou/GalleriesForYouScreen.tsx [hunk 1]
```diff
@@ -56,6 +77,12 @@ export const GalleriesForYou: React.FC<GalleriesForYouProps> = ({ location }) =>
       info={screen({ context_screen_owner_type: OwnerType.galleriesForYou })}
     >
       <Flex>
+        {!!visualizeLocation && (
+          <Text ml={6} color="red">
+            Location: {JSON.stringify(location)}
+          </Text>
+        )}
```
Comment (olerichter00) [inline src/app/Scenes/GalleriesForYou/GalleriesForYouScreen.tsx:84]: Adding this for better debugging.

File: src/app/Scenes/GalleriesForYou/GalleriesForYouScreen.tsx [hunk 2]
```diff
@@ -47,6 +53,21 @@ export const GalleriesForYou: React.FC<GalleriesForYouProps> = ({ location }) =>
...
+  useEffect(() => {
+    if (partners.length || hasRefetched) return
+
+    refetch({
+      ...queryParams,
+      includePartnersWithFollowedArtists: false,
+    })
+
+    setHasRefetched(true)
+  })
```
Comment (MounirDhahri) [inline src/app/Scenes/GalleriesForYou/GalleriesForYouScreen.tsx:69]: even if you're confident about this piece of code working always I would suggest adding a dependency array here

Comment (MounirDhahri) [reply src/app/Scenes/GalleriesForYou/GalleriesForYouScreen.tsx:69]: you never know how the code will grow

File: src/app/Scenes/GalleriesForYou/GalleriesForYouScreen.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/GalleriesForYou/GalleriesForYouScreen.tsx:84]: good idea

**Review by MounirDhahri — APPROVED** [review]
LGTM - only a small comment about the `useEffect`

File: src/app/Scenes/GalleriesForYou/GalleriesForYouScreen.tsx (hunk 2 above)
Comment (olerichter00) [reply src/app/Scenes/GalleriesForYou/GalleriesForYouScreen.tsx:69]:
```suggestion
  }, [partners])
```

**Pull request merged**